### PR TITLE
add Ab:RBD analysis scripts

### PR DESCRIPTION
**Pull request by glass-w**
This PR adds analysis scripts used for Ab:RBD systems

TODO

- [x] add analysis scripts
- [x] add yaml files for plotting
- [x] update README

### CODE REVIEW

File: 02_analysis_scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+## Analysis
+Below are exmaples of how to use analysis scripts to analyse the dataset for F@H data in [DOI link]. It is recommened to run each of these scripts in separate directories.
```
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:2]:
```suggestion
Below are examples of how to use analysis scripts to analyse the dataset for F@H data in [DOI link]. It is recommended to run each of these scripts in separate directories.
```

File: 02_analysis_scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+**Note**: these scripts were developed as part of a time sensive project, as a result they are not memory efficient.
+
+### Contact analysis
+`analyze_antibody_rbd_contacts.py` can be used to generate a series of `.pkl` files containing an N x M x No. frames array of distances between heavy atoms at the RBD:Ab interface.
```
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:7]: can you define N and M here?

File: 02_analysis_scripts/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+For example, analysing the F@H `17341` project (S2H97:Ab):
+
+```python
+python analyze_antibody_rbd_contacts.py -mutant WT -pdb_file system.pdb -traj_path path/to/traj/ -fah_project_code 17341 -chain_info_file chain_info_S2H97_RBD.yaml -run_number 0 -discard_time 100
+```
+* `system.pdb` should contain no solvent.
```
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:14]:
```suggestion
* `system.pdb` is a PDB of the RBD:antibody without waters, but with ions.
```

File: 02_analysis_scripts/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+**Note**: these scripts were developed as part of a time sensive project, as a result they are not memory efficient.
+
+### Contact analysis
+`analyze_antibody_rbd_contacts.py` can be used to generate a series of `.pkl` files containing an N x M x No. frames array of distances between heavy atoms at the RBD:Ab interface.
+
+For example, analysing the F@H `17341` project (S2H97:Ab):
+
+```python
+python analyze_antibody_rbd_contacts.py -mutant WT -pdb_file system.pdb -traj_path path/to/traj/ -fah_project_code 17341 -chain_info_file chain_info_S2H97_RBD.yaml -run_number 0 -discard_time 100
```
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:12]: can you define what `-chain_info_file` and `-run_number` and `-discard_time` correspond to?

File: 02_analysis_scripts/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+* `path/to/contacts/` should be a path containing the `.pkl` files generated by `analyze_antibody_rbd_contacts.py`.
+* The `-metadata` flag should point to a `.pkl` metadata file generated by `analyze_antibody_rbd_contacts.py`.
+
+### Plotting
+Running `count_ab_rbd_interactions.py` produces one `.pkl` file per antibody chain (e.g. count_chain1.pkl and count_chain2.pkl). These can then be used in the `make_hmaps.py` script.
+
+```python
+python make_hmaps.py -data count_chain1.pkl count_chain2.pkl -save_path path/to/save/plots -ab_name S2H97 -custom_residues_file s2h97_ab_custom_residues.yaml
```
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:31]: can you define what `-custom_residues_file` corresponds to?

File: 02_analysis_scripts/README.md (hunk 4 above)
Comment (zhang-ivy) [reply 02_analysis_scripts/README.md:12]: And also specify that there are 5 RUNs for each project, so we ran this command on each of the 5 RUNs separately. And that we used a discard time of 100 for all analysis jobs.

File: 02_analysis_scripts/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+## Analysis
+Below are exmaples of how to use analysis scripts to analyse the dataset for F@H data in [DOI link]. It is recommened to run each of these scripts in separate directories.
+
+**Note**: these scripts were developed as part of a time sensive project, as a result they are not memory efficient.
+
+### Contact analysis
+`analyze_antibody_rbd_contacts.py` can be used to generate a series of `.pkl` files containing an N x M x No. frames array of distances between heavy atoms at the RBD:Ab interface.
+
+For example, analysing the F@H `17341` project (S2H97:Ab):
```
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:9]:
```suggestion
For example, analysing the F@H `17347` project (RBD:S309):
```

Comment (zhang-ivy) [reply 02_analysis_scripts/README.md:9]: Can you add a line above this that says something like: 
Project numbers:
RBD:S309 - 17341
RBD:S2H97 - 17347

File: 02_analysis_scripts/README.md (hunk 4 above)
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:12]:
```suggestion
python analyze_antibody_rbd_contacts.py -mutant WT -pdb_file system.pdb -traj_path path/to/traj/ -fah_project_code 17341 -chain_info_file chain_info_S309_RBD.yaml -run_number 0 -discard_time 100
```

File: 02_analysis_scripts/README.md (hunk 5 above)
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:31]:
```suggestion
python make_hmaps.py -data count_chain1.pkl count_chain2.pkl -save_path path/to/save/plots -ab_name S2H97 -custom_residues_file s309_ab_custom_residues.yaml
```

**Review by zhang-ivy — CHANGES_REQUESTED** [review]

**Comment by zhang-ivy** [comment]
Also, just noticed that in the main README, the manifest section says `02_analysis/` -- could you either change the name of the directory in this PR or change the name in the main README? up to you which name you want to keep!

**Comment by zhang-ivy** [comment]
Also, I noticed main README `02_analysis` part says `analysis script` instead of `scripts`.. could you update that as well?

**Comment by zhang-ivy** [comment]
Also, I noticed the main README has a typo near the BAOAB langevin integrator. Need to replace `78` with the reference (Leimkuhler and Matthews, 2013)[https://doi.org/10.1063/1.4802990]

**Comment by zhang-ivy** [comment]
I can make all the main README changes if you want, just let me know when you're done fixing this PR!

**Comment by glass-w** [comment]
@zhang-ivy I've addressed your comments and updated the main README to have `02_analysis_scripts`

File: 02_analysis_scripts/README.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+antibody_chains:
+  chain_1: A
+  chain_2: B
+rbd_chain:
+  chain_1: R
+glycan_chain:
+  chain_1: X
+```
+where `A`, `B`, `R`, and `X` are the names of the indicated chain in the PDB file.
+* The `-discard_time` flag specifies how much time to discard for a particular CLONE before starting contact analysis.
```
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:31]:
```suggestion
* The `-discard_time` flag specifies how much time (ns) to discard for a particular CLONE before starting contact analysis.
```

File: 02_analysis_scripts/README.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+For this project there were a total of 5 RUNs, so the above script was ran on each RUN separately. Additioanlly a `-discard_time` of 100 ns was applied
```
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:33]:
```suggestion
For this project there were a total of 5 RUNs, so the above script was ran on each RUN separately. Additionally a `-discard_time` of 100 ns was applied
```

File: 02_analysis_scripts/README.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+* `-project_number` corresponds to the F@H project number.
+
+### Plotting
+Running `count_ab_rbd_interactions.py` produces one `.pkl` file per antibody chain (e.g. count_chain1.pkl and count_chain2.pkl). These can then be used in the `make_hmaps.py` script.
+
+```python
+python make_hmaps.py -data count_chain1.pkl count_chain2.pkl -save_path path/to/save/plots -ab_name S2H97 -custom_residues_file s2h97_ab_custom_residues.yaml
+```
+
+* The `-custom_residues_file` should point to a YAML file containing the system type (e.g. WT). See `s2h97_ab_custom_residues.yaml` for an example.
```
Comment (zhang-ivy) [inline 02_analysis_scripts/README.md:62]: I think you mean:
```suggestion
* The `-custom_residues_file` should point to a YAML file containing the RBD residues for which to compute the % contribution to total interface close contacts. See `s2h97_ab_custom_residues.yaml` for an example.
```

**Review by zhang-ivy — APPROVED** [review]

**Pull request merged**